src/Generation/GirTool/GenerateCommand.cs

```diff
@@ -137,15 +137,14 @@ private static (DeserializedInput, DeserializedInput, DeserializedInput) LoadRep
         return (linuxRepositories!, macosRepositories!, windowsRepositories!);
     }
 
-    private static DeserializedInput DeserializeInput(string? searchPath, string[] input)
+    private static DeserializedInput DeserializeInput(string platformName, string? searchPath, string[] input)
     {
-        if (searchPath is null)
-            return DeserializedInput.Empty();
+        var repositoryResolver = GetRepositoryResolver(platformName, searchPath);
 
         var inputRepositories = input
-            .Select(x => Path.Join(searchPath, x))
-            .Where(File.Exists)
-            .Select(x => new FileInfo(x).OpenRead().DeserializeGirInputModel())
+            .Select(fileName => repositoryResolver.ResolveRepository(fileName))
+            .Where(inputRepository => inputRepository != null)
+            .Select(inputRepository => inputRepository!)
```

Comment:
```suggestion
```

This is obsolete.

Reply:
This was needed to avoid nullability errors later as the compiler thinks the type of `inputRepositories` is `List<IRepository?>` rather than `List<IRepository>` without this line. But your suggestion to use `OfType` means this won't be needed.